### DISCUSSION TITLE
Set max deep upload in policy.

### DISCUSSION THREAD

**Discussion by logneh4** [supabase/supabase#12676] [Questions] [answered]
I want the user to be able to upload a file but not be able to create folders.

I have policy:
`(storage.filename(name) = auth.uid()::text || '.png')`

Im trying use `array_ndims((storage.dirnname(name)))` but its not working.

**Comment by GaryAustin1** [comment, marked as answer]
`dirnname`   What is that?   `storage.foldername(name)` is probably what you wanted.   

You should be able to use array_length on that.  https://www.postgresql.org/docs/current/arrays.html

**Reply by logneh4** [reply]
storage.diname(name): `Error updating policy: failed to update pg.policies with the given ID: function storage.dirname(text) does not exist` 
dirname: `Error updating policy: failed to update pg.policies with the given ID: column "dirname" does not exist`

**Reply by GaryAustin1** [reply]
Yes, I said I've never seen a function named dirnname or dirname involved with Supabase storage.  Is this a function you created?  
Also there is no column call dirname in storage.objects.

Here are docs on the storage functions and storage RLS:  https://supabase.com/docs/guides/storage/access-control

**Reply by logneh4** [reply]
My bad, I pasted the wrong code (I had `foldername` and pasted `dirname`) and then was wondering what's going on...
I did it this way and it works:
`((array_length(storage.foldername(name), 1) = 0) OR (array_length(storage.foldername(name), 1) IS NULL))`
